### ISSUE TITLE
section-title: updated "see all" variant

### ISSUE THREAD

**Issue by rbascue** [eBay/skin#1728]
The section title component is being updated to replace the arrow icon button with a link button pinned to the right of the screen. The button defaults to have a label of "See all".
DOD: Section title styling has been updated to match Playbook specs.

[Figma component](https://www.figma.com/file/zEBdEhbonrBOGzZ0fXzWvM/?node-id=54324%3A83599)

[Playbook page](https://playbook.corp.ebay.com/PlaybookApp#/executional-guidelines/product-design-system/patterns/section-titles)

![image-2022-04-20-11-05-09-499](https://user-images.githubusercontent.com/35155833/164303996-3aaef4d6-740d-4372-9b64-4be3f1b834fc.png)
![image-2022-04-20-11-05-23-562](https://user-images.githubusercontent.com/35155833/164303999-e0ef5501-931c-452a-bc94-94675cd07235.png)
![image-2022-04-20-11-05-36-464](https://user-images.githubusercontent.com/35155833/164304000-a53c5333-bbf3-42e6-b0b7-7a8da4278489.png)
![image-2022-04-20-11-05-47-464](https://user-images.githubusercontent.com/35155833/164304005-b5aa25d5-7c4d-400a-a32a-bd85bdd9ebbd.png)
![image-2022-04-20-11-05-55-564](https://user-images.githubusercontent.com/35155833/164304009-f0dda510-52e7-44aa-a1db-72f40a5dc60a.png)

**Comment by ianmcburnie** [comment]
There has been some back and forth on this and various permutations, so for the benefit of the dev working on this, here are the versions currently supported:

![Screen Shot 2022-04-20 at 12 11 05 PM](https://user-images.githubusercontent.com/38065/164305226-384dd19f-478d-4cf7-8a0e-6e7a2a2effab.png)

The version above is hidden from docs and deprecated, scheduled for removal in v14. It was recently superseded by the version below, but seem like it's reign will now be short-lived:

![Screen Shot 2022-04-20 at 12 10 42 PM](https://user-images.githubusercontent.com/38065/164305661-11fbe24e-d168-498f-8f5e-e54342d54831.png)

**Comment by agliga** [comment]
Should be for next major version. Will be breaking change.

**Comment by ArtBlue** [comment]
@rbascue , currently, the section title heading is an anchor. There is no mention of this in the issue, but can I assume we're removing the anchor and using plain text for the heading, and only the link on the right will be a link? The playbook element descriptions seem to indicate it too, but I wanted to verify before implementing the change.

<img width="895" alt="image" src="https://user-images.githubusercontent.com/1675667/188907115-27a1f2b5-deed-4600-89a3-2dee0c48b969.png">

**Comment by ArtBlue** [comment]
@rbascue , also, the figma link does not show the actual component (I can't get to it either from inside); it just has images of various components and their statuses. Can you give me the link to the figma that I can inspect to get layout details to be used?

**Comment by rbascue** [comment]
Hi @ArtBlue, here is a link to the component in the library. https://www.figma.com/file/zEBdEhbonrBOGzZ0fXzWvM/eBay-Design-System?node-id=78903%3A90028

**Comment by ArtBlue** [comment]
@rbascue , thanks for that. One more Q - at the moment, storybook has a few more variations that look to be obsolete. Can you verify they should be removed? https://opensource.ebay.com/skin/storybook/?path=/story/section-title-rtl--heading 

Specifically, the medium and giant variations appear to be obvious candidates for removal. I just want to clean things up as much as possible. What do you think? Anything else that sticks out at you for removal?

**Comment by ArtBlue** [comment]
@rbascue , also, it looks like both the small and large variations use the small overflow icon (3x13). Is this intentional? We do have a large overflow (4x18), but going with the large would mean changing the button from 32x32 to 40x40 for the large variation.

**Comment by ArtBlue** [comment]
@rbascue , there are also a few style variations applied (in code) on the base styles and breakpoints over 601px. I don't see those specifications based on breakpoints in Figma. Do all screen widths now use the same styling? Should we remove those breakpoints overrides?

**Comment by agliga** [comment]
So from what I understand is we should do it like this (we can let Randy confirm). But this should unblock you imo.

- There is a large and small section in figma. For small it should be below 601 and large should be over. 
<img width="775" alt="Screen Shot 2022-09-08 at 11 27 14 AM" src="https://user-images.githubusercontent.com/1755269/189198155-0043cbbb-7cbc-460e-9444-45ad8f0cafee.png">
If we aren't using those, then we should remove it. Lets sync up as much as we can with figma currently and remove the old stuff.
If you feel there's too many things changing, let us know what it is. In dialog we removed a lot of the large/small styles since they now are the same pretty much. 

- Whatever variation we have and is not in figma should be removed.

- I believe design system is defaulting to small icons across the board. Lets use the small icon for now and switch it if needed.

**Comment by rbascue** [comment]
@ArtBlue @agliga Yes, large would be for 601+ and we're sticking with the smaller icon button for both sizes.
